packages/flutter/lib/src/material/dropdown.dart

```diff
@@ -906,12 +907,12 @@ class DropdownButton<T> extends StatefulWidget {
 class _DropdownButtonState<T> extends State<DropdownButton<T>> with WidgetsBindingObserver {
   int _selectedIndex;
   _DropdownRoute<T> _dropdownRoute;
+  Orientation _lastOrientation;
 
   @override
   void initState() {
     super.initState();
     _updateSelectedIndex();
-    WidgetsBinding.instance.addObserver(this);
```

Comment:
if we never add ourself as an observer we can also remove the removal in the dispose method (and stop implementing the WidgetsBindingObserver interface.